app/assets/javascripts/components/common/text_input.jsx

```diff
@@ -1,116 +1,138 @@
-import React from 'react';
-import createReactClass from 'create-react-class';
+import React, { useRef } from 'react';
 import PropTypes from 'prop-types';
 import InputHOC from '../high_order/input_hoc.jsx';
 import Conditional from '../high_order/conditional.jsx';
 
-const TextInput = createReactClass({
-  displayName: 'TextInput',
+const TextInput = ({
+  value,
+  value_key,
+  editable,
+  label,
+  placeholder,
+  spacer,
+  valueClass,
+  p_tag_classname,
+  inline,
+  type,
+  max,
+  maxLength,
+  focus,
+  onBlur,
+  onClick,
+  append,
+  onKeyDown,
+  _value,
+  invalid,
+  onChange,
+  onFocus,
+  id,
+  children
+}) => {
+  const inputRef = useRef(null);
 
-  propTypes: {
-    value: PropTypes.any,
-    value_key: PropTypes.string,
-    editable: PropTypes.bool,
-    label: PropTypes.string,
-    placeholder: PropTypes.string,
-    spacer: PropTypes.string,
-    valueClass: PropTypes.string,
-    p_tag_classname: PropTypes.string,
-    inline: PropTypes.bool,
-    type: PropTypes.string,
-    max: PropTypes.string,
-    maxLength: PropTypes.string,
-    focus: PropTypes.func,
-    onBlur: PropTypes.func,
-    onClick: PropTypes.func,
-    append: PropTypes.node,
-    onKeyDown: PropTypes.func,
-    _value: PropTypes.any,
-    // validation: Regex used by Conditional
-    // required: bool used by Conditional
-  },
+  const onKeyDownHandler = (e) => {
+    if (!onKeyDown) return;
+    onKeyDown(e.keyCode, inputRef.current);
+  };
 
-  onKeyDown(e) {
-    if (!this.props.onKeyDown) { return; }
-    this.props.onKeyDown(e.keyCode, this.refs.inputbox);
-  },
+  let labelContent;
+  const usedSpacer = spacer || ': ';
 
-  dateChange(date) {
-    const value = date ? date.format('YYYY-MM-DD') : '';
-    return this.props.onChange({ target: { value } });
-  },
+  if (label) {
+    labelContent = label + usedSpacer;
+  }
 
-  render() {
-    let label;
-    const spacer = this.props.spacer || ': ';
+  let usedValueClass = 'text-input-component__value ';
+  if (valueClass) {
+    usedValueClass += valueClass;
+  }
 
-    if (this.props.label) {
-      label = this.props.label + spacer;
+  if (editable) {
+    let labelClass = '';
+    let inputClass = inline ? 'inline' : '';
+    if (invalid) {
+      labelClass += 'red';
+      inputClass += ' invalid';
```

Comment:
Can be simplified:

```
const labelClass = invalid ? 'red' : '';
const inputClass =`${inline ? 'inline' : ''} ${invalid ? 'invalid' : ''}`
```

This way one does not have to keep in mind trailing whitespaces required before concatenating another class.

For inputClass there's only two variables for now and the above string interpolation is fine. But if there were to be more variables then one can do:

```
const inputClass = [
  inline ? 'inline' : '',
  invalid ? 'invalid' : '',
  // other conditional class names
].join(' ');
```

Reply:
Done, Thanks for reviewing @AmitJoki